src/httpclient.c

```diff
@@ -423,7 +427,7 @@ h2o_httpclient_head_cb on_connect(h2o_httpclient_t *client, const char *errstr,
     }
 
     *_method = h2o_iovec_init(req.method, strlen(req.method));
-    *url = *(strcmp(req.method, "CONNECT") == 0 ? &req.connect_to : (h2o_url_t *)client->data);
+    *url = *(strcmp(req.method, "CONNECT") == 0 ? req.connect_to : (h2o_url_t *)client->data);
```

Comment:
I'm not sure if this change is correct. Would you mind trying `-x ... -m CONNECT` to see if it works as expected?

To give an example, with the h2o configuration file being something like below, and run `-x 127.0.0.1:8080 -m CONNECT http://www.fastly.com:80/`, then it should create a TCP tunnel to www.fastly.com:80. But I think this code creates a tunnel to the TCP proxy through that TCP proxy.

```yaml
listen:
  port: 8080
hosts:
  default:
    /:
      proxy.connect:
        - ":+*"
```

Reply:
@kazuho You're right:

```
$ ./h2o-httpclient -x 127.0.0.1:8080 -m CONNECT http://www.fastly.com:80/
HTTP/1.1 200 OK
Connection: close
Server: h2o/2.3.0-DEV@d491bb0

GET / HTTP/1.1
Host: www.fastly.com

HTTP/1.1 404 File Not Found
Connection: keep-alive
Content-Length: 9
Server: h2o/2.3.0-DEV@d491bb0
content-type: text/plain; charset=utf-8

not found

socket closed by peer
```

It should be fixed in [90c8912c](https://github.com/h2o/h2o/pull/2691/commits/90c8912c43334eb3670c50e31650a795d6db2dd2).  Could you please check it?